packages/server/src/internals/procedure.ts

```diff
@@ -121,8 +121,10 @@ export class Procedure<TInputContext, TContext, TInput, TOutput> {
       return async (nextOpts?: { ctx: TContext }) => {
         const res = await wrapCallSafe(() =>
           fn({
-            ...(opts as any),
-            ...(nextOpts as any),
+            ctx: nextOpts ? nextOpts.ctx : opts.ctx,
+            type: opts.type,
+            path: opts.path,
+            rawInput: opts.rawInput,
```

Comment:
Is this change actually needed?

Reply:
This change is _not_ strictly needed for this PR, but I think this is cleaner / clearer so we don't need to cast `opts` and `nextOpts` to `any`. I also think it's best to not pass params that are not in the types so clients don't begin to [rely on them just because they are there](https://www.hyrumslaw.com/).